test/e2e/specs/specs-i18n/wp-i18n__editor.ts

```diff
@@ -214,85 +214,101 @@ const translations: Translations = {
+		describe( 'Set up', function () {
+			it( `Change UI language to ${ locale }`, async function () {
+				await Promise.all( [
+					page.waitForNavigation( { url: '**/home/**', waitUntil: 'load' } ),
+					page.goto( DataHelper.getCalypsoURL( '/' ) ),
+				] );
 
-			await Promise.all( [
-				page.waitForNavigation( { url: '**/home/**', waitUntil: 'load' } ),
-				page.goto( DataHelper.getCalypsoURL( '/' ) ),
-			] );
-		} );
+				const changeUILanguageFlow = new ChangeUILanguageFlow( page );
+				await changeUILanguageFlow.changeUILanguage( locale as LanguageSlug );
 
-		it( 'Start new post', async () => {
-			const newPostFlow = new NewPostFlow( page );
-			await newPostFlow.newPostFromNavbar();
+				await Promise.all( [
+					page.waitForNavigation( { url: '**/home/**', waitUntil: 'load' } ),
+					page.goto( DataHelper.getCalypsoURL( '/' ) ),
+				] );
+			} );
 
-			gutenbergEditorPage = new GutenbergEditorPage( page );
+			it( 'Start new post', async function () {
+				const newPostFlow = new NewPostFlow( page );
+				await newPostFlow.newPostFromNavbar();
+			} );
 		} );
 
-		describeSkipNoTranslations.each( localeTranslations.blocks )(
+		describe.each( translations[ locale ].blocks )(
 			'Translations for block: $blockName',
-			( block ) => {
+			function ( block ) {
 				let frame: Frame;
+				let gutenbergEditorPage: GutenbergEditorPage;
 
-				beforeAll( async () => {
-					frame = await gutenbergEditorPage.getEditorFrame();
+				const blockTimeout = 10 * 1000;
+
+				it( 'Insert test block', async function () {
+					gutenbergEditorPage = new GutenbergEditorPage( page );
 					await gutenbergEditorPage.addBlock( block.blockName, block.blockEditorSelector );
 				} );
 
-				it( 'Render block content translations', async () => {
+				it( 'Render block content translations', async function () {
+					frame = await gutenbergEditorPage.getEditorFrame();
+					// Ensure block contents are translated as expected.
 					await Promise.all(
-						block.blockEditorContent.map( ( content ) =>
-							frame.waitForSelector( `${ block.blockEditorSelector } ${ content }` )
+						block.blockEditorContent.map( ( content: any ) =>
+							frame.waitForSelector( `${ block.blockEditorSelector } ${ content }`, {
+								timeout: blockTimeout,
+							} )
 						)
 					);
 				} );
 
-				it( 'Render block title translations', async () => {
+				it( 'Render block title translations', async function () {
 					await gutenbergEditorPage.openSettings();
 					await frame.click( block.blockEditorSelector );
-					await Promise.race( [
-						frame.waitForSelector( `${ block.blockEditorSelector }.is-selected` ),
-						frame.click( '.block-editor-block-parent-selector__button' ),
-					] );
 
+					// Ensure the block is highlighted.
+					await frame.waitForSelector(
+						`:is( ${ block.blockEditorSelector }.is-selected, ${ block.blockEditorSelector }.has-child-selected)`,
+						{ timeout: blockTimeout }
```

Comment:
Instead of using `Promise.race`, this now uses the [`:is`](https://playwright.dev/docs/selectors#selecting-elements-matching-one-of-the-conditions) pseudo-CSS selector. This selector will match on a list of selectors passed in as argument.

This is necessary as in some cases (eg. Contact Form block) there exists sub-blocks, and if these child blocks are selected then the attributed added is `has-child-selected` instead of `is-selected`.